src/Benchmarks/Data/EfDb.cs

```diff
@@ -49,9 +57,10 @@ public Task<World> LoadSingleQueryRow()
             for (var i = 0; i < count; i++)
             {
                 var id = _random.Next(1, 10001);
-                var result = await _dbContext.World.AsTracking().FirstAsync(w => w.Id == id);
+                var result = await _firstWorldQuery(_dbContext, id);
+
+                _dbContext.Attach(result).Property("RandomNumber").CurrentValue = _random.Next(1, 10001);
```

Comment:
I think the best solution of this is to use `Find()` and to fix `Find()` to take advantage of compiled queries (I have created https://github.com/aspnet/EntityFramework/issues/7248 for this). Once we do this code can be changed to:

``` 
C#
var id = _random.Next(1, 10001);
var result = await _dbContext.World.Find(id);
var entry = _dbContext.Entry(result);
if (entry.State == EntityState.Detached)
{
    entry.State = EntityState.Unchanged;
}
entry.Property("RandomNumber").CurrentValue = _random.Next(1, 10001);
```